src/core/components/PictureUploader.vue

```diff
@@ -0,0 +1,148 @@
+        @click="uploadImage" round flat size="1rem" />
+      <q-btn v-if="hasPicture && disablePictureEdition" color="primary" round flat icon="save_alt" size="1rem"
+        type="a" :href="pictureDlLink(pictureLink)" target="_blank" />
+    </div>
+    </div>
+</template>
+
+<script>
+import { Cookies } from 'quasar';
+import get from 'lodash/get';
+import Users from '@api/Users';
+import cloudinary from '@api/Cloudinary';
+import { NotifyPositive, NotifyNegative } from '@components/popup/notify';
+import { removeDiacritics } from '@helpers/utils';
+
+export default {
+  name: 'PictureUploader',
+  props: {
+    userProfile: { type: Object, default: () => {} },
+    background: { type: String, default: 'client-background' },
+    refreshPicture: { type: Function, default: () => {} },
+  },
+  data () {
+    return {
+      disablePictureEdition: true,
+      croppa: {},
+      loadingImage: false,
+      fileChosen: false,
+    };
+  },
+  computed: {
+    hasPicture () {
+      return !!this.pictureLink;
+    },
+    pictureUploadUrl () {
+      return `${process.env.API_HOSTNAME}/users/${this.userProfile._id}/cloudinary/upload`;
+    },
+    pictureLink () {
+      return get(this.userProfile, 'picture.link') || null;
+    },
+    canvasColor () {
+      return this.background === 'client-background' ? '#EEE' : '#FFEDDA';
+    },
+    userRole () {
+      if (get(this.userProfile, 'role.client')) return this.userProfile.role.client.name;
+      else if (get(this.userProfile, 'role.vendor')) return this.userProfile.role.vendor.name;
+      return 'no_role';
+    },
+  },
+  methods: {
+    async uploadImage () {
+      try {
+        if (this.hasPicture && !this.fileChosen) {
+          await cloudinary.deleteImageById({ id: this.userProfile.picture.publicId });
+        }
+        this.loadingImage = true;
+        const blob = await this.croppa.promisedBlob('image/jpeg', 0.8);
+        const data = new FormData();
+        data.append('_id', this.userProfile._id);
+        data.append('role', this.userRole);
+        data.append('fileName', `photo_${this.userProfile.identity.firstname}_${this.userProfile.identity.lastname}`);
+        data.append('Content-Type', blob.type || 'application/octet-stream');
+        data.append('picture', blob);
+
+        await this.$axios.post(
+          this.pictureUploadUrl,
+          data,
+          { headers: { 'content-type': 'multipart/form-data', 'x-access-token': Cookies.get('alenvi_token') || '' } }
+        );
+        await this.refreshPicture();
+        this.closePictureEdition();
+        NotifyPositive('Modification enregistrée');
+      } catch (e) {
+        console.error(e);
+        NotifyNegative('Erreur lors de la modification.');
+      } finally {
+        this.loadingImage = false;
+      }
+    },
+    async deleteImage () {
+      try {
+        if (this.userProfile.picture && this.userProfile.picture.publicId) {
```

Comment:
get(this.userProfile, 'picture.publicId')

Reply:
done